src/AzureFunctions.PowerShell.OpenTelemetry.SDK/Logging/NewFunctionsOpenTelemetryLogger.cs

```diff
@@ -20,9 +20,10 @@ public class NewFunctionsOpenTelemetryLogger : PSCmdlet
     {
         protected override void ProcessRecord()
         {
-            if (string.IsNullOrEmpty(Environment.GetEnvironmentVariable(SDKConstants.FunctionsOpenTelemetryEnvironmentVariableName))) 
+            if (!FunctionsEnvironmentHelper.IsFunctionsEnvironmentVariableEnabled() && !FunctionsEnvironmentHelper.HasWarnedAboutEnvironmentVariable()) 
```

Comment:
Also, I suggest reverting the order: check `HasWarnedAboutEnvironmentVariable` first, as it seems to be _significantly_ cheaper.

Reply:
Done